marlowe-runtime/indexer/Language/Marlowe/Runtime/Indexer/Database/PostgreSQL/CommitBlocks.hs

```diff
@@ -437,69 +452,72 @@ withdrawTxToWithdrawalTxInRows BlockHeader{..} MarloweWithdrawTransaction{..} =
       , fromIntegral createTxIx
       )
 
-prepareParams :: [MarloweBlock] -> QueryParams
+prepareParams :: [MarloweBlock] -> ([ContractTxOutParty], QueryParams)
 prepareParams blocks =
-  ( V.fromList blockIdRows
-  , V.fromList blockSlotRows
-  , V.fromList blockNoRows
-  , V.fromList txOutTxIdRows
-  , V.fromList txOutTxIxRows
-  , V.fromList txOutBlockIdRows
-  , V.fromList txOutAddressRows
-  , V.fromList txOutLovelaceRows
-  , V.fromList txOutAssetTxIdRows
-  , V.fromList txOutAssetTxIxRows
-  , V.fromList txOutAssetBlockIdRows
-  , V.fromList txOutAssetPolicyIdRows
-  , V.fromList txOutAssetNameRows
-  , V.fromList txOutAssetQuantityRows
-  , V.fromList contractTxOutTxIdRows
-  , V.fromList contractTxOutTxIxRows
-  , V.fromList contractTxOutBlockIdRows
-  , V.fromList contractTxOutPayoutScriptHashRows
-  , V.fromList contractTxOutContractRows
-  , V.fromList contractTxOutStateRows
-  , V.fromList contractTxOutRolesCurrencyRows
-  , V.fromList createTxOutTxIdRows
-  , V.fromList createTxOutTxIxRows
-  , V.fromList createTxOutSlotNoRows
-  , V.fromList createTxOutBlockIdRows
-  , V.fromList createTxOutBlockNoRows
-  , V.fromList createTxOutMetadataRows
-  , V.fromList applyTxTxIdRows
-  , V.fromList applyTxCreateTxIdRows
-  , V.fromList applyTxCreateTxIxRows
-  , V.fromList applyTxSlotNoRows
-  , V.fromList applyTxBlockIdRows
-  , V.fromList applyTxBlockNoRows
-  , V.fromList applyTxInvalidBeforeRows
-  , V.fromList applyTxInvalidHereafterRows
-  , V.fromList applyTxMetadataRows
-  , V.fromList applyTxInputTxIdRows
-  , V.fromList applyTxInputTxIxRows
-  , V.fromList applyTxInputsRows
-  , V.fromList applyTxOutputTxIxRows
-  , V.fromList payoutTxOutTxIdRows
-  , V.fromList payoutTxOutTxIxRows
-  , V.fromList payoutTxOutBlockIdRows
-  , V.fromList payoutTxOutRolesCurrencyRows
-  , V.fromList payoutTxOutRoleRows
-  , V.fromList withdrawalTxInTxIdRows
-  , V.fromList withdrawalTxInSlotNoRows
-  , V.fromList withdrawalTxInBlockIdRows
-  , V.fromList withdrawalTxInBlockNoRows
-  , V.fromList withdrawalTxInPayoutTxIdRows
-  , V.fromList withdrawalTxInPayoutTxIxRows
-  , V.fromList withdrawalTxInCreateTxIdRows
-  , V.fromList withdrawalTxInCreateTxIxRows
-  , V.fromList invalidApplyTxTxIdRows
-  , V.fromList invalidApplyTxInputTxIdRows
-  , V.fromList invalidApplyTxInputTxIxRows
-  , V.fromList invalidApplyTxBlockIdRows
-  , V.fromList invalidApplyTxErrorRows
-  , V.fromList contractTxOutTagTagRows
-  , V.fromList contractTxOutTagTxIdRows
-  , V.fromList contractTxOutTagTxIxRows
+  ( concat parties
```

Comment:
I had to run these queries separately - apparently GHC cannot handle tuples larger than 74 items.

Reply:
🤯